onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -323,6 +323,9 @@ def __init__(self, path_or_bytes, sess_options=None, providers=None, provider_op
         else:
             raise TypeError("Unable to load from type '{0}'".format(type(path_or_bytes)))
 
+        if providers is None:
```

Comment:
lines 326-328 should not be added. it will negate the enforcement check in line 359-363 below.

Reply:
@jywu-msft Sorry that I missed your comments somehow. Just changed it back.